### DISCUSSION TITLE
关于websocketio的一些疑问

### DISCUSSION THREAD

**Discussion by GX88** [midwayjs/midway#4035] [Q&A]
我现在使用koa与midwayjs/socketio。我现在使用了全局中间件，去验证每次连接的认证是否有问题。我现在有两个不同的websocket接口，一个需要认证，一个是开放的。我怎么可以根据地址的一些特征去判断是否需要认证呢。也就是想问我怎么能拿到每次请求连接的地址。感谢

**Comment by czy88840616** [comment]
OnWSConnection 方法有 socket 和 request 两个参数，request 即为 http 的 request。

**Reply by GX88** [reply]
我打印了ctx.request，发现url为“”url: '/socket.io/?EIO=4&transport=polling&t=P6S8Chi'“”，我想获取的地址应该是/admin/ws。不然我不知道怎么去做判断
```javascript
socket.value = io("http://127.0.0.1:8001/admin/ws",{
    auth: {
      token: '123'
    }
  })
```

**Reply by czy88840616** [reply]
/socket.io/ 是默认的 socket.io的 path，没问题

**Reply by czy88840616** [reply]
![image](https://github.com/user-attachments/assets/875e2477-6205-42a7-b15a-525b50aae69a)
这里文档有描述。

**Reply by GX88** [reply]
可能是我没有反馈清楚，我想问的是@WSController("/test")，前端得地址`http://127.0.0.1:8001/test`；我想在服务器判断是否是/test，然后判断是否需要认证😂

**Reply by czy88840616** [reply]
不行的吧，你请求发出后已经是那个 namespace 地址，你没法在外面判断是否进那个 namespace。

```ts
const io = require('socket.io')(server);

const nsp = io.of('/my-namespace');
nsp.on('connection', (socket) => {
  // 这里的 socket 对象是针对 '/my-namespace' 的
});
```

**Reply by GX88** [reply]
那我说一下我的场景，看您有解决方案吗。我现在想在后台管理中实时获取服务器的运行状态等，我需要长连接，我定了一个地址/ws的地址。所以这个接口肯定需要认证，也就是保证连接需要携带token。
但是后续我可能需要其他长连接的场景，例如/test，我不需要认证。
所以我有了全局中间件，我需要在中间件根据不同的长连接接口，去做不同接口的处理逻辑，是否需要认证。目前我的问题在于中间件无法判断是哪个长连接接口。

**Reply by czy88840616** [reply]
![image](https://github.com/user-attachments/assets/a8fe0667-d391-4694-bea6-1574056888ef)
走 qurey ？
